docs/architecture/ADR-003-Non-interactive-defaults.md

```diff
@@ -0,0 +1,698 @@
+# ADR 003: Non-interactive Defaults, Wrapped Transactions, and Subtree Root Message Inclusion Checks
+
+NOTE: Unlike normal tendermint/cosmos ADRs, this ADR isn't for deciding on whether or not we will implement non-interactive defaults. The goal of this document is to help reviewers and future readers understand what non-interactive defaults are, the considerations that went into the initial implementation, and how it differs from the original specs.
+
+The exact approach taken by the initial implementation, however, is certainly up for scrutiny.
+
+## Intro
+
+pls see the [original specs](https://github.com/celestiaorg/celestia-specs/blob/master/src/rationale/message_block_layout.md), from which this ADR paraphrases heavily.
+
+Currently, when checking for message inclusion, validators recreate the share commitment from the messages in the block and compare those with what are signed over in the `MsgPayForData` transactions also in that block. If any commitment is not found in one of the PFD transactions, or if there is a commitment that doesn't have a corresponding message, then they reject that block.
+
+While this functions as a message inclusion check, the light client has to assume that 2/3's of the voting power is honest in order to be assured that both the messages they are interested in and the rest of the messages paid for in that block are actually included. In order to have this property, we need a block validity rule where:
+
+> **All share commitments included in `MsgPayForData` must consist only of subtree roots of the data square.**
+
+The main issue with that requirement is that users must know the relevant subtree roots before they sign, which is problematic considering that if the block is not organized perfectly, the subtree roots will include data unknown to the user at the time of signing.
+
+To fix this, the spec outlines the “non-interactive default rules”. These involve a few additional **default but optional** message layout rules that enables the user to follow the above block validity rule, while also not interacting with a block producer. Commitments to messages can consist entirely of sub tree roots of the data hash, and for those sub tree roots to be generated only from the message itself (so that the user can sign something “non-interactively”).
+
+NOTE: THE STATEMENT BELOW IS MODIFIED FROM THE SPEC
```

Comment:
nit: I do not think we need to capitalize this (given that we deprecated the spec and it has a huge warning in the readme). But don't feel strongly about it either.

Reply:
removed, e706becb357c3042f8ed3971b17cccf26def1dd7 as this was mainly a comment for reviewers